ipapython/ipaldap.py

```diff
@@ -723,18 +721,121 @@ def __init__(self, ldap_uri, start_tls=False, force_schema_updates=False,
             If true, attributes are decoded to Python types according to their
             syntax.
         """
+        if debug and debug.lower() == "on":
+            ldap.set_option(ldap.OPT_DEBUG_LEVEL, 255)
+        if cacert is not None:
+            ldap.set_option(ldap.OPT_X_TLS_CACERTFILE, cacert)
+
         self.ldap_uri = ldap_uri
         self._start_tls = start_tls
         self._force_schema_updates = force_schema_updates
         self._no_schema = no_schema
         self._decode_attrs = decode_attrs
+        self.cacert = cacert
 
         self.log = log_mgr.get_logger(self)
         self._has_schema = False
         self._schema = None
 
         self._conn = self._connect()
 
+        with self.error_handler():
+            if demand_cert:
+                ldap.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, True)
+                self.conn.set_option(ldap.OPT_X_TLS_DEMAND, True)
+
+            if sasl_nocanon:
+                self.conn.set_option(ldap.OPT_X_SASL_NOCANON, ldap.OPT_ON)
+
+            if start_tls:
+                self.conn.start_tls_s()
+
+    @staticmethod
+    def IPAdmin_init(host='', port=389, cacert=None, debug=None,
```

Comment:
This should be a function called `IPAdmin` rather than a static method of `LDAPClient`, that way you don't have to modify any of the existing code which calls `IPAdmin`.